Tests/Tracking/CustomerIOTest.swift

```diff
@@ -262,4 +262,89 @@ class CustomerIOTest: UnitTest {
 
         waitForExpectations()
     }
+
+    // MARK: track
+
+    func test_track_givenSdkNotInialized_expectFailureResult() {
+        customerIO = CustomerIO(credentialsStore: nil, sdkConfig: SdkConfig(), identifyRepository: nil,
+                                keyValueStorage: nil)
+
+        let expect = expectation(description: "Expect to complete track")
+        customerIO.track(name: String.random) { result in
+            guard case .failure(let error) = result else { return XCTFail() }
+            guard case .notInitialized = error else { return XCTFail() }
+
+            XCTAssertFalse(self.identifyRepositoryMock.mockCalled)
+
+            expect.fulfill()
+        }
+
+        waitForExpectations()
+    }
+
+    func test_track_expectCallRepository() {
+        let givenEventName = String.random
+        let givenEventData = TrackEventData.random()
+
+        identifyRepositoryMock.trackEventClosure = { actualEventName, actualEventData, _, _, onComplete in
+            XCTAssertEqual(givenEventName, actualEventName)
+            XCTAssertEqual(givenEventData, actualEventData.value as! TrackEventData)
+
+            onComplete(Result.success(()))
+        }
+
+        let expect = expectation(description: "Expect to complete track")
+        customerIO.track(name: givenEventName, data: givenEventData) { result in
+            expect.fulfill()
+        }
+
+        waitForExpectations()
```

Comment:
```suggestion
        waitForExpectations()
        
        XCTAssertEqual(identifyRepositoryMock.trackEventCalledCount, 1)
```

Since as of now, the test function doesn't test that the repository did in fact get called.